### ISSUE TITLE
Importing a mocked ESM module dynamically will throw an error in node 16.17.0

### ISSUE THREAD

**Issue by adamraine** [testdouble/testdouble.js#498]
### Description

Importing a mocked ESM module dynamically will throw an error in node 16.17.0 but not 16.16.0.

### Issue

Mocking an esm module will result on an error

### Environment

- [x] `node -v` output: v16.17.0
- [x] `yarn --version` output: 1.22.19
- [x] `yarn list testdouble` version: testdouble@3.16.5

#### Code-fenced Examples

```
❯❯❯ node --loader=testdouble script.js
(node:83826) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:83826) DeprecationWarning: Obsolete loader hook(s) supplied and will be ignored: getFormat, getSource
node:internal/errors:477
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_LOADER_CHAIN_INCOMPLETE]: "testdouble 'load'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
    at new NodeError (node:internal/errors:387:5)
    at ESMLoader.load (node:internal/modules/esm/loader:616:13)
    at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:464:11)
    at async link (node:internal/modules/esm/module_job:67:21) {
  code: 'ERR_LOADER_CHAIN_INCOMPLETE'
}
```

```js
import * as td from 'testdouble';
await td.replaceEsm('fs', {});
await import('fs');

```

**Comment by searls** [comment]
I haven't used `td.replaceEsm` in earnest, but it would be unusual to import a mocked module from the same code listing.

I would have expected:

```js
import * as td from 'testdouble';
const fs = await td.replaceEsm('fs', {});
```

And then the subject under test would be loaded and it would in turn `import` the mocked module. As per [this example in the docs](https://github.com/testdouble/testdouble.js/blob/main/docs/7-replacing-dependencies.md#nodejs). 

As a result I'm not sure this is supported. Can you elaborate if it seems like I'm off-base?

**Comment by connorjclark** [comment]
The bug actually came from usage of my fork, not qubbile mainline. Sorry about that. This was fixed by #74 – simply rebasing the latest code fixed it for us.
